tests/test_cli.py

```diff
@@ -26,11 +26,67 @@ def test_simple_workflow(app, db, tmpdir):
 
     source = os.path.join(os.path.dirname(__file__), 'fixtures', 'source')
 
+    # Create a location to use
     result = runner.invoke(cmd, [
-        'location', 'tmp', 'file://' + tmpdir.strpath, '--default'
+        'location', 'create',
+        'tmp', 'file://' + tmpdir.strpath, '--default'
     ], obj=script_info)
     assert 0 == result.exit_code
 
+    # Create the same location (check idempotent)
+    result = runner.invoke(cmd, [
+        'location', 'create',
+        'tmp', 'file://' + tmpdir.strpath, '--default'
+    ], obj=script_info)
+    assert 0 == result.exit_code
+    assert "already exists" in result.output
+
+    # Create a second one as default to check tmp is not default anymore
+    result = runner.invoke(cmd, [
+        'location', 'create',
+        'aux', 'file://' + tmpdir.strpath, '--default'
+    ], obj=script_info)
+    assert 0 == result.exit_code
+
+    # List locations and check the default is correct
+    result = runner.invoke(cmd, [
+        'location', 'list'
+    ], obj=script_info)
+    assert 0 == result.exit_code
+
+    created_locations = result.output.split('\n')
+
+    # tmp is not default
+    assert "tmp" in created_locations[0]
+    assert "as default False" in created_locations[0]
+    # aux is default
+    assert "aux" in created_locations[1]
+    assert "as default True" in created_locations[1]
+
+    # Set tmp back as default
+    result = runner.invoke(cmd, [
+        'location', 'set-default', 'tmp'
+    ], obj=script_info)
+    assert 0 == result.exit_code
+
+    # List locations and check the default is correct
+    result = runner.invoke(cmd, [
+        'location', 'list'
+    ], obj=script_info)
+    assert 0 == result.exit_code
+
+    created_locations = result.output.split('\n')
+    # tmp is default
+    assert "tmp" in created_locations[0]
+    assert "as default True" in created_locations[0]
+    # aux is not default
+    assert "aux" in created_locations[1]
+    assert "as default False" in created_locations[1]
+
```

Comment:
*I already gave my approval and should have provided those yesterday, so feel free to ignore my comments. I try to be consistent and want to get those remarks out there*: 

The already existing tests here were testing against the cli output, so I see the appeal in following in their footsteps. However, given they interact with the DB, testing what's in the DB is probably more accurate and valuable than what is printed out. But I am not sure where the line is for tests that run fast, are valuable and are convenient to write. Maybe what we have here is a good enough comprise...

Reply:
I kind of agree with you. Due to time constraints (and the fact that it works -  tested manually) we I'd go with it and open an issue. Since the test against db should be done not only for my new commands but for all the CLI ones (buckets, etc.)